### PR TITLE
feat(auditlog): Add audit logs for ondemand spend

### PR DESCRIPTION
**Pull request by MeredithAnya**
Adding audit logs for when ondemand spend changes.

@tkaemming wondering if this okay to add for the Sentry codebase even though really only is for billing.

### CODE REVIEW

**Comment by tkaemming** [comment]
This is a good question and a hard one to answer. I'm inclined to say it's not a great idea from a correctness standpoint … but at the same time it doesn't really hurt anything and the work that would be required to expose these internals to the other repository doesn't really seem worth it unless we plan on doing this more often in the future.

I'm not sure if @mattrobenolt and/or @dcramer have any better ideas or a different perspective?

**Comment by ehfeng** [comment]
My thoughts:

1. Adding constants doesn't hurt the open source community.
2. The work and complexity of exposing internals slows us down from getting out a valuable, requested feature to our users/customers.

Therefore, this pattern is fine. Plus, we can always double back and re-factor it later.

File: src/sentry/models/auditlogentry.py [hunk 1]
```diff
@@ -243,4 +246,7 @@ def get_note(self):
         elif self.event == AuditLogEntryEvent.RULE_REMOVE:
             return 'removed rule "%s"' % (self.data['label'], )
 
+        elif self.event == AuditLogEntryEvent.SET_ONDEMAND:
+            return 'changed on-demand max spend to $%d' % (self.data['ondemand'] / 100, )
```
Comment (tkaemming) [inline src/sentry/models/auditlogentry.py]: Note that this is going to only show integer values (just dollars), not decimals. I'm not sure if that's relevant.

**Comment by ehfeng** [comment]
@MeredithAnya small tweak: I'd suggest moving to the 90s range for the `AuditLogEntryEvent`, just for clarity.

**Review by tkaemming — APPROVED** [review]
Nobody's articulated a strong argument against doing this, so I'm going ahead and approving it.

If we continue this pattern we should reevaluate, though.

**Comment by tkaemming** [comment]
> I'd suggest moving to the 90s range for the AuditLogEntryEvent, just for clarity.

This is a good idea too.

File: src/sentry/models/auditlogentry.py (hunk 1 above)
Comment (ehfeng) [reply src/sentry/models/auditlogentry.py]: On-demand budgets can only be in dollars. Because we're evil, greedy people 🤑

**Pull request merged**